### PR TITLE
Add pins for Lerdge X and K.

### PR DESCRIPTION
**Pull request by xC0000005**
### Requirements

* Filling out this template is required. Pull Requests without a clear description may be closed at the maintainers' discretion.

### Description

Adds pins definitions for Lerdge's X and K boards.

### Benefits

Allows one, with the help of an encryption script and some Arduino board definitions, to compile Marlin for the Lerdge X and K.

### Related Issues

#12624

### CODE REVIEW

File: Marlin/src/pins/stm32/pins_LERDGE_K.h [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+#define SDSS               PC11
+#define LED_PIN            PC7         //Alive
+#define PS_ON_PIN          -1
+#define KILL_PIN           -1
+#define PWR_LOSS           -1         //Power loss / nAC_FAULT
+
+#define SCK_PIN   PC12
+#define MISO_PIN  PC8
+#define MOSI_PIN  PD2
+#define SS_PIN    PC11
```
Comment (thinkyhead) [inline Marlin/src/pins/stm32/pins_LERDGE_K.h]: Are these not provided by the platform? They should be the same on all STM32F4.

**Comment by tpruvot** [comment]
small typo in comment about the Lerdge K SoC ref.. 
https://drive.google.com/file/d/1xvG3y87g02m15PyCyuEv8WJfHlUJhvIu/view?usp=sharing

**Comment by thisiskeithb** [comment]
> Allows one, with the help of an encryption script

Do you have those?

**Comment by tpruvot** [comment]
just one, the K (ok oops lol)

**Comment by xC0000005** [comment]
Yes. I will add a link to it shortly.

**Pull request merged**